zipline/utils/tradingcalendar_nyse.py

```diff
@@ -0,0 +1,269 @@
+from datetime import time
+from itertools import chain
+
+from dateutil.relativedelta import (
+    MO,
+    TH,
+)
+from pandas import (
+    date_range,
+    DateOffset,
+    Timestamp,
+)
+from pandas.tseries.holiday import(
+    AbstractHolidayCalendar,
+    GoodFriday,
+    Holiday,
+    nearest_workday,
+    sunday_to_monday,
+    USLaborDay,
+    USPresidentsDay,
+    USThanksgivingDay,
+)
+from pandas.tseries.offsets import Day
+from pytz import timezone
+
+from zipline.utils.tradingcalendar import ExchangeCalendar
+
+# Useful resources for making changes to this file:
+# http://www.nyse.com/pdfs/closings.pdf
+# http://www.stevemorse.org/jcal/whendid.html
+
+MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY = range(7)
+
+US_EASTERN = timezone('US/Eastern')
+NYSE_OPEN = time(9, 31)
+NYSE_CLOSE = time(16)
+NYSE_STANDARD_EARLY_CLOSE = time(13)
+
+# Closings
+USNewYearsDay = Holiday(
+    'New Years Day',
+    month=1,
+    day=1,
+    # When Jan 1 is a Sunday, NYSE observes the subsequent Monday.  When Jan 1
+    # Saturday (as in 2005 and 2011), no holiday is observed.
+    observance=sunday_to_monday
+)
+USMemorialDay = Holiday(
+    # NOTE: The definition for Memorial Day is incorrect as of pandas 0.16.0.
+    # See https://github.com/pydata/pandas/issues/9760.
+    'Memorial Day',
+    month=5,
+    day=25,
+    offset=DateOffset(weekday=MO(1)),
+)
+USMartinLutherKingJrAfter1998 = Holiday(
+    'Dr. Martin Luther King Jr. Day',
+    month=1,
+    day=1,
+    # The NYSE didn't observe MLK day as a holiday until 1998.
+    start_date=Timestamp('1998-01-01'),
+    offset=DateOffset(weekday=MO(3)),
+)
+USIndependenceDay = Holiday(
+    'July 4th',
+    month=7,
+    day=4,
+    observance=nearest_workday,
+)
+Christmas = Holiday(
+    'Christmas',
+    month=12,
+    day=25,
+    observance=nearest_workday,
+)
+
+# Half Days
+MonTuesThursBeforeIndependenceDay = Holiday(
+    # When July 4th is a Tuesday, Wednesday, or Friday, the previous day is a
+    # half day.
+    'Mondays, Tuesdays, and Thursdays Before Independence Day',
+    month=7,
+    day=3,
+    days_of_week=(MONDAY, TUESDAY, THURSDAY),
+    start_date=Timestamp("1995-01-01"),
+)
+FridayAfterIndependenceDayExcept2013 = Holiday(
+    # When July 4th is a Thursday, the next day is a half day (except in 2013,
+    # when, for no explicable reason, Wednesday was a half day instead).
+    "Fridays after Independence Day that aren't in 2013",
+    month=7,
+    day=5,
+    days_of_week=(FRIDAY,),
+    observance=lambda dt: None if dt.year == 2013 else dt,
+    start_date=Timestamp("1995-01-01"),
+)
+USBlackFridayBefore1993 = Holiday(
```

Comment:
Early closings were at 2 PM until 1993, so we need a separate calendar for the different times.